### PR TITLE
Feature/youtube display

### LINKED ISSUE #10: Single Movie Display - stytling
Add styling to single movie display 
Add backdrop photo as background image!

### LINKED ISSUE #25: As a user, I should be able to see a trailer of the movie when I display the details page
When a specific movie is clicked, the details page should load with movie details and trailer.

### PR DESCRIPTION
**Pull request by ashton-huxtable**
PR Template:

Is this a fix or a feature?
- [ ] Bug fix
- [x] Feature
- [ ] Refactor

Describe the feature - fix - refactor:
Add fetch requests for movie trailers and modify existing movie details fetch.
Add MovieTrailer component
Add basic styling to movie details page 

How was this tested?
- [ ] open index.html
- [ ] localhost
- [ ] terminal
- [x] React Server

Where should the reviewer start - test:
Details component and MovieTrailer component

Additional comments:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "@testing-library/jest-dom": "^5.14.1",
     "@testing-library/react": "^11.2.7",
     "@testing-library/user-event": "^12.8.3",
+    "prop-types": "^15.7.2",
```
Comment (shayan-golafshani) [inline package.json:9]: Nice job adding prop types

File: public/index.html [hunk 2]
```diff
@@ -3,6 +3,11 @@
   <head>
     <meta charset="utf-8" />
     <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <link rel="preconnect" href="https://fonts.googleapis.com">
```
Comment (shayan-golafshani) [inline public/index.html:6]: Nice fonts

File: src/components/Details/Details.js [hunk 3]
```diff
@@ -8,21 +9,36 @@ class Details extends React.Component {
...
-          })
-          .catch(err => {
-            console.error(err, 'Error caught in get request for movie details')
-            this.setState({errorMessage: 'Something went wrong, please try again later 😔'})
-          });
+
+        let getMovie = (endpoint) => {
+           let fetchData =  
+           endpoint ?
+           fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${this.props.selectedMovie.id}/${endpoint}`)
```
Comment (shayan-golafshani) [inline src/components/Details/Details.js:21]: Next time, I'll follow your approach and not refactor in my head.

File: src/components/Details/Details.js [hunk 4]
```diff
@@ -39,22 +55,44 @@ class Details extends React.Component {
             title,
             } = this.state.movie;
 
+            const backgroundStyle = {
```
Comment (shayan-golafshani) [inline src/components/Details/Details.js:58]: Nice inline styling!

File: src/components/MovieTrailer/MovieTrailer.css [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    align-content: center;
+  }
+  
+  .video-responsive iframe {
+    left: 0;
+    top: 0;
+    bottom: 0;
+    right: 0;
+    object-fit: contain;
+    position: absolute;
```
Comment (shayan-golafshani) [inline src/components/MovieTrailer/MovieTrailer.css:19]: Let's try making sure it's relative to the parent

**Review by shayan-golafshani — APPROVED** [review]
looks great

File: src/components/App/App.css [hunk 6]
```diff
@@ -6,19 +6,28 @@
...
   color: #FFF;
   text-align: center;
-  font-size: 2rem;;
 }
 
 h1 {
-  /* margin-top: 0.5rem;; */
+font-family: 'Monoton', cursive;
+font-size: 3rem;;
+align-self: center;
```
Comment (kaylagordon) [inline src/components/App/App.css:32]: It's hard to tell in this UI, but are these indented correctly?

File: src/components/Details/Details.js [hunk 7]
```diff
@@ -8,21 +9,36 @@ class Details extends React.Component {
         this.state = {
             errorMessage: '',
             movie: {},
-            movieTrailerUrl: {},
+            movieTrailers: [],
```
Comment (kaylagordon) [inline src/components/Details/Details.js:12]: Would it make more sense for this property to be inside the `movie` object?

File: src/components/Details/Details.js (hunk 3 above)
Comment (kaylagordon) [inline src/components/Details/Details.js:21]: Consider moving your API calls into a separate file.

File: src/components/Details/Details.js [hunk 8]
```diff
@@ -8,21 +9,36 @@ class Details extends React.Component {
...
+                }
+            })
+            .catch(err => {
+                console.error(err, 'Error caught in get request for movie details')
+                this.setState({errorMessage: 'Something went wrong, please try again later 😔'})
+            });
+            }
+
+          Promise.all([getMovie(), getMovie('videos')])
+          .then(promise => console.log(promise));
```
Comment (kaylagordon) [inline src/components/Details/Details.js:41]: Remember to remove all console.logs!

File: src/components/MovieCard/MovieCard.css [hunk 9]
```diff
@@ -13,7 +14,8 @@ img {
 .card-rating {
     margin: 0.1rem;
     font-size: 2rem;
-    text-align: center;;
+    text-align: center;
+    /* text-transform: uppercase; */
```
Comment (kaylagordon) [inline src/components/MovieCard/MovieCard.css:18]: Remove commented code.

File: src/components/MovieTrailer/MovieTrailer.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import PropTypes from 'prop-types';
+import './MovieTrailer.css'
+
+const MovieTrailer = ({ movieKey }) => (
+  <div className="video-responsive">
+    <iframe
+      // width="853"
+      // height="480"
+      src={`https://www.youtube.com/embed/${movieKey}`}
```
Comment (kaylagordon) [inline src/components/MovieTrailer/MovieTrailer.js:10]: Nice!

File: src/components/MovieTrailer/MovieTrailer.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+      frameBorder="0"
+      allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+      allowFullScreen
+      title="Embedded youtube"
+    />
+  </div>
+);
+
+MovieTrailer.propTypes = {
+  movieKey: PropTypes.string.isRequired
```
Comment (kaylagordon) [inline src/components/MovieTrailer/MovieTrailer.js:20]: Great work using `PropTypes` here! I like that you've added the `isRequired` property too!

File: src/components/Details/Details.js [hunk 12]
```diff
@@ -8,21 +9,36 @@ class Details extends React.Component {
...
+           endpoint ?
+           fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${this.props.selectedMovie.id}/${endpoint}`) 
+           : fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${this.props.selectedMovie.id}`);
+
+        return fetchData
+                .then(response => response.json())
+                .then(data => {
+                if(data.movie) {
+                let movie = data.movie
+                this.setState({movie})
```
Comment (kaylagordon) [inline src/components/Details/Details.js:29]: These two lines should be indented in.

**Review by kaylagordon — COMMENTED** [review]
Overall, great work! No need to make any of the changes I recommended, but feel free to read through my comments and implement any changes you see fit. Let me know if you have any questions!

**Pull request merged**